### PR TITLE
Docs cyberark

### PR DESCRIPTION
**Pull request by farchide**
providing CyberArk integration documentations

### CODE REVIEW

File: docs/docs/configuration/secrets.md [hunk 1]
```diff
@@ -29,13 +23,47 @@ To do so, export a `username=secret` environment variable where the value of tha
...
 To support this, the `secret` must not be set in the connector's configuration file **and** there musn't be any **Azure key vault** configured in the main configuration file.
 
-# Put the secrets in Azure Key Vaults
+
+# Put the secrets in CyberArk
+**CyberArk** (http://www.cyberark.com) Application Access Manager for DevOps provides a secrets management solution tailored specifically to the unique requirements of native-cloud and DevOps environments. The solution manages secrets and credentials used by non-human identities including DevOps and PaaS tools, and containers. **Prancer** validation framework supports secrets to be read from the **CyberArk**. Here is the steps:
+
+1. DevOps Engineer starts the **Prancer** validation framework.
+2. **Prancer** validation framework is integrated with CyberArk to retrieve secrets
+    a. CyberArk Agent has to be installed and configured on the respective servers (containers) and the firewall ports has to be opened
```
Comment (crazycodr) [inline docs/docs/configuration/secrets.md]: Have to be opened

Comment (crazycodr) [reply docs/docs/configuration/secrets.md]: This part is weird, I dont feel it should be written like that. I think that part should be part of the configuration of the service not part of the explanation how prancer resolves the secrets

File: docs/docs/configuration/secrets.md [hunk 2]
```diff
@@ -29,13 +23,47 @@ To do so, export a `username=secret` environment variable where the value of tha
...
+
+# Put the secrets in CyberArk
+**CyberArk** (http://www.cyberark.com) Application Access Manager for DevOps provides a secrets management solution tailored specifically to the unique requirements of native-cloud and DevOps environments. The solution manages secrets and credentials used by non-human identities including DevOps and PaaS tools, and containers. **Prancer** validation framework supports secrets to be read from the **CyberArk**. Here is the steps:
+
+1. DevOps Engineer starts the **Prancer** validation framework.
+2. **Prancer** validation framework is integrated with CyberArk to retrieve secrets
+    a. CyberArk Agent has to be installed and configured on the respective servers (containers) and the firewall ports has to be opened 
+3. The **Prancer** validation framework runs the cyberArk agent cli and connects to the CyberArk safe to fetch the Password for the account.
+4. CyberArk passes the retrieved password to the modules that need the secret during the validation process
+5. if any errors - due to CyberArk not installed or system errors or password retrieving error, the **Prancer** validation framework checks for the other ways to retireve the secret
```
Comment (crazycodr) [inline docs/docs/configuration/secrets.md]: If any errors occur (due to CyberArk not being installed properly or any other errors) then ...

File: docs/docs/configuration/secrets.md [hunk 3]
```diff
@@ -29,13 +23,47 @@ To do so, export a `username=secret` environment variable where the value of tha
 
 To support this, the `secret` must not be set in the connector's configuration file **and** there musn't be any **Azure key vault** configured in the main configuration file.
 
-# Put the secrets in Azure Key Vaults
+
+# Put the secrets in CyberArk
```
Comment (crazycodr) [inline docs/docs/configuration/secrets.md]: Putting secrets in

File: docs/docs/configuration/secrets.md [hunk 4]
```diff
@@ -29,13 +23,47 @@ To do so, export a `username=secret` environment variable where the value of tha
...
+```
+[VAULT]
+type = cyberark
+CA_OBJECT = 'object name'
+CA_SAFE = True
+CA_EXE = 'Path to the cyberark executable'
+CA_APPID = 'APP ID used for storing the object name'
+```
+
+# Put the secrets in Azure Key Vault
```
Comment (crazycodr) [inline docs/docs/configuration/secrets.md]: Putting secrets in

File: docs/docs/configuration/secrets.md [hunk 5]
```diff
@@ -1,18 +1,12 @@
-**Prancer** can leverage secrets from different sources such as an **Azure key vault** when required.
-
-This concept only applies to the **Azure** connector that you will see a little later. If you do not want to store your SPN secrets in files anywhere, then a good alternative will be to use an **Azure key vault**.
-
-# Limitations
-
-It is important to understand that **Azure key vaults** are only available in interactive mode because the SPN password to access the key vault must be typed on the console when running the tests. This doesn't prevent you from using secrets as there are other ways to do so. Read on!
+In order to connect to the back-end API system, **Prancer** can leverage secrets from different sources when required.
```
Comment (crazycodr) [inline docs/docs/configuration/secrets.md]: to connect to a back-end system

File: docs/docs/configuration/secrets.md [hunk 6]
```diff
@@ -29,13 +23,47 @@ To do so, export a `username=secret` environment variable where the value of tha
 
 To support this, the `secret` must not be set in the connector's configuration file **and** there musn't be any **Azure key vault** configured in the main configuration file.
 
-# Put the secrets in Azure Key Vaults
+
+# Put the secrets in CyberArk
+**CyberArk** (http://www.cyberark.com) Application Access Manager for DevOps provides a secrets management solution tailored specifically to the unique requirements of native-cloud and DevOps environments. The solution manages secrets and credentials used by non-human identities including DevOps and PaaS tools, and containers. **Prancer** validation framework supports secrets to be read from the **CyberArk**. Here is the steps:
```
[the remark this answers is not in the document]
Comment (crazycodr) [reply docs/docs/configuration/secrets.md]: Build tools, PaaS tools and containers

**Comment by crazycodr** [comment]
I found other things but I cant seem to post them from the mobile web interface

File: docs/docs/configuration/secrets.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (farchide) [reply docs/docs/configuration/secrets.md]: This is from Cyber Ark website. i would like to keep it as is

File: docs/docs/configuration/secrets.md (hunk 1 above)
Comment (farchide) [reply docs/docs/configuration/secrets.md]: let's have it here until we find a better place to put it in

**Review by crazycodr — APPROVED** [review]

**Pull request merged**